src/config/index.ts

```diff
@@ -161,9 +161,8 @@ export class ConfigUtil {
             `-prefs=build.warn_data_percentage=${this._warnPercentage}`,
             `-prefs=runtime.tools.avr-gcc.path=${this._idePath}/hardware/tools/avr`,
             `-prefs=runtime.tools.avrdude.path=${this._idePath}/hardware/tools/avr`,
-            `-prefs=runtime.tools.arduinoOTA.path=${this._idePath}/hardware/tools/avr`,
-            vscode.window.activeTextEditor.document.fileName
-        ].concat(this._compileOptions);
+            `-prefs=runtime.tools.arduinoOTA.path=${this._idePath}/hardware/tools/avr`
+        ].concat(this._compileOptions, vscode.window.activeTextEditor.document.fileName);

```

Comment:
if this is a real problem, then the upload should also put file flags to the last one.